### ISSUE TITLE
popups of some xwayland apps do not pop up anymore

### ISSUE THREAD

**Issue by ElJeffe** [hyprwm/Hyprland#6616]
### Regression?

Yes

### System Info and Version

<details> <summary>System/Version info</summary>

```sh
Hyprland, built from branch main at commit fabc30df52ab5d2c369fc8acd4ff909a6ba3b8ac
```

</details>

### Description

Popup menus stopped showing in OrcaSlicer and BambuStudio since commit 2566d818848b58b114071f199ffe944609376270
Here I click on the 'seam position' setting:
[Kooha-2024-06-21-10-59-32.webm](https://github.com/hyprwm/Hyprland/assets/54236/760bab5a-2f62-4327-95f6-cee6ca729141)

### How to reproduce

Start OrcaSlicer and click on a setting with a popup.

### Crash reports, logs, images, videos

Log when clicking the setting:
[popup.log](https://github.com/user-attachments/files/15926597/popup.log)

**Comment by vaxerski** [comment]
...wayland apps? The commit relates to xwayland.

**Comment by ElJeffe** [comment]
> ...wayland apps? The commit relates to xwayland.

Yes xwayland. I must have mistyped.

**Comment by UltraBlackLinux** [comment]
any updates on this? Really frustrating, and window rules are unable to fix that either. Making the menus unfocusable and forcing the parent window to keep the focus doesn't work at all, as if the window rules didn't exist.

**Comment by MightyPlaza** [comment]
> any updates on this? Really frustrating, and window rules are unable to fix that either. Making the menus unfocusable and forcing the parent window to keep the focus doesn't work at all, as if the window rules didn't exist.

it's a regression from https://github.com/hyprwm/Hyprland/commit/2566d818848b58b114071f199ffe944609376270, you can just revert it and build hyprland

**Comment by UltraBlackLinux** [comment]
I'm hoping to see a fix for this somewhat soon, at least a toggle to undo that commit at runtime...

I currently have to juggle between both the latest and that commit to get Bambu Studio and Bitwig working. Very frustrating since I either cannot slice anything or xwayland crashes when using bitwig...

**Comment by vaxerski** [comment]
isnt this fixed? as a dupe of #6698

**Comment by Vanderscycle** [comment]
I still get the same issue as of version `V.0.41.2`
```
❯ hyprctl version
Hyprland, built from branch main at commit 918d8340afd652b011b937d29d5eea0be08467f5  (flake.lock: update).
Date: 2024-06-25
Tag: v0.41.2, commits: 918d8340afd652b011b937d29d5eea0be08467f5
```
